client/src/EntriesList.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { Entry, readEntries } from './data';
+import { Link } from 'react-router-dom';
+import { FaPencilAlt } from 'react-icons/fa';
+
+export function EntriesList() {
+  const [entries, setEntries] = useState<Entry[]>([]);
+
+  useEffect(() => {
+    async function read() {
+      try {
+        const response = await readEntries();
```

Comment:
You should let the user know that the page is loading. Otherwise if there are network delays the user will think something is broken and will leave your site.